java/google/registry/flows/FlowMetadata.java

```diff
@@ -0,0 +1,24 @@
+package google.registry.flows;
+
+import com.google.auto.value.AutoValue;
+import google.registry.model.ImmutableObject;
+
+/** Object to allow setting and retrieving metadata information in flows. */
```

Comment:
Since @AutoValue is immutable, unlike SessionMetadata, this object doesn't directly allow "setting" metadata information.  I'd say something more like "Object to hold metadata specific to a particular execution of a flow."  That also captures that this metadata is tied to a single flow execution, rather than a flow *class*, and contrasts it with SessionMetadata which has session lifetime.